### PR TITLE
Review and integrate open250

### PR DESCRIPTION
**Pull request by charlesh88**
#250 and open250 address three front-end issues:
- #185 - Allow the tree pane to handle its layout better when bundles are not included, like search.
- #157 - Hide the Create button by default while in mobile, but allow devs to re-display it if needed by including a bundle.
- #260 - Fix a regression bug that was preventing the view switcher menu from appearing when invoked from the main object view area.

**Author checklist:**
- Changes address original issue? Y
- Unit tests included and/or updated with changes? Y
- Command line build passes? Y
- Expect to pass code review? Y

### CODE REVIEW

File: platform/commonUI/browse/res/templates/browse/object-header.html [hunk 1]
```diff
@@ -19,12 +19,14 @@
...
-<div class='object-header'>
-    <span class='type-icon ui-symbol'>{{type.getGlyph()}}</span>
-    <!--span class='type-name mobile-important-hide'>{{type.getName()}}</span-->
-    <span class="l-elem-wrapper l-flex">
-        <span ng-if="parameters.mode" class='action'>{{parameters.mode}}</span>
-        <span class='title-label'>{{model.name}}</span>
-        <mct-representation key="'menu-arrow'" mct-object='domainObject'></mct-representation>
-    </span>
-</div>
+<!--<div class='object-header'> Moved up to browse-object.html and frame.html. TO-DO: make sure this is Ok!-->
```
Comment (VWoeltjen) [inline platform/commonUI/browse/res/templates/browse/object-header.html]: Has this TODO been addressed?

File: platform/commonUI/general/res/sass/_archetypes.scss [hunk 2]
```diff
@@ -97,23 +97,54 @@
 .l-flex-col {
     @include display-flex;
     @include flex-wrap(nowrap);
+    //@include flex(1 1 auto !important);
     .flex-elem {
+        //@include test(green, 0.1);
```
Comment (VWoeltjen) [inline platform/commonUI/general/res/sass/_archetypes.scss]: Is there value/utility to having these commented-out lines here? In general prefer to remove unused code

File: platform/commonUI/general/res/sass/controls/_controls.scss [hunk 3]
```diff
@@ -214,7 +214,7 @@ label.checkbox.custom {
 
     .l-elem-wrapper {
         //@include test(#66f, 0.2);
-        @include justify-content(flex-start);
+        //@include justify-content(flex-start); // Commented out 11/6/15 open250
```
Comment (VWoeltjen) [inline platform/commonUI/general/res/sass/controls/_controls.scss]: As above, do we need to keep this line commented-out, or should it be removed?

File: platform/commonUI/general/res/sass/search/_search.scss [hunk 4]
```diff
@@ -20,70 +20,50 @@
...
-	//display: flex;
-	@include webkitProp(flex-direction, column);
-	//flex-direction: column;
-	height: 100%;
 
 	.search-bar {
 		$textInputHeight: 19px; // This is equal to the default value, 19px
 		$iconEdgeM: 4px;
 		$iconD: $treeSearchInputBarH - ($iconEdgeM*2);
+        //@include test(red, 0.1);
```
Comment (VWoeltjen) [inline platform/commonUI/general/res/sass/search/_search.scss]: Looks like another dead comment

File: platform/search/res/templates/search.html [hunk 5]
```diff
@@ -62,47 +47,30 @@
...
-            <span class="title-label">Loading...</span>
-        </div>
-
+    <div class="search-results flex-elem grows vscroll"
+         ng-class="{ off: !(loading || results.length > 0), loading: loading }">
+
+        <!--ng-hide="(ngModel.input === '' || ngModel.input === undefined)"-->
+
+        <!--<div class="hint" ng-show="results.length === 0">No results found</div>-->
+        <!--<div class="hint" ng-show="results.length > 0">Displaying {{ results.length }} result<span ng-show="results.length > 1">s</span></div>-->
```
Comment (VWoeltjen) [inline platform/search/res/templates/search.html]: Should these comments be kept, uncommented, or removed?

**Comment by charlesh88** [comment]
Uh, if you start looking for commented dead code in my sass, you're going to continually trip over it. This hasn't come up as a general issue previously, although Pete called me out on it in one particular instance a little while ago.

Let me go through and kill the comments in the sass in this branch, I'll push up a new version shortly. I'd really like the sass in #250 for something I'm doing in #279, so I'm highly motivated to get this integrated.

File: platform/commonUI/browse/res/templates/browse/object-header.html (hunk 1 above)
Comment (charlesh88) [inline platform/commonUI/browse/res/templates/browse/object-header.html]: Yes, it has. Will remove to-do.

File: platform/search/res/templates/search.html (hunk 5 above)
Comment (charlesh88) [inline platform/search/res/templates/search.html]: Sorry, meant to remove that. Will do so.

**Comment by VWoeltjen** [comment]
Thanks - I'm trying to be more diligent about catching little things like that during code review (it's the only time they really get caught). Reassign to me when you're ready; I'll keep an eye out to complete the merge.

**Comment by charlesh88** [comment]
No prob, and thanks. Moving forward I'll ensure that commented stuff is stamped out before doing pull reqs.

**Comment by VWoeltjen** [comment]
Hm, so running this branch (open250) locally, I get a strange display (left pane starts collapsed):

open274

If I manually adjust the splitter next to the tree, the tree displays correctly, but I still have no inspector.

Wonder if this is a more severe case of #169? (I'm viewing on my little laptop screen)

**Comment by charlesh88** [comment]
Re. "...I get a strange display...": was totally able to duplicate this in the Chrome mobile emulator: this is #169. I'm going to make fixing that a priority so you (and others with small laptops) won't be impacted by this.

**Comment by charlesh88** [comment]
Ok, just pushed an updated version of open250 - should be completely cleaned up.

**Comment by VWoeltjen** [comment]
### Reviewer Checklist

1. Changes appear to address issue? Y
2. Appropriate unit tests included? N/A &ast;
3. Code style and in-line documentation are appropriate? Y
4. Commit messages meet standards? Y

&ast; changes to CSS & templates only, out of scope for automated testing

**Comment by charlesh88** [comment]
Awesome, thanks!

**Pull request merged**